packages/builder/src/builderStore/store/index.js

```diff
@@ -185,6 +186,26 @@ const setCurrentScreen = store => screenName => {
   })
 }
 
+const deleteScreens = store => (screens, pageName = null) => {
+  if (!(screens instanceof Array)) {
+    screens = [screens]
+  }
+  store.update(state => {
+    if (pageName == null) {
+      pageName = state.pages.main.name
+    }
+    for (let screen of screens) {
+      state.screens = state.screens.filter(c => c.name !== screen.name)
+      // Remove screen from current page as well
+      state.pages[pageName]._screens = state.pages[pageName]._screens.filter(
```

Comment:
Using `pageName` here might mean that if the user was on the unathenticated page, then went to backend and deleted a screen it wouldn't actually remove it from the "page" structure. I'd replace this with just `state.pages.main._screens` but double check that still works. The `pageName` var is just "main" or "unauthenticated" I think, but the screens we want to delete are always under "main",

Reply:
I only use page name here as I used the stuff that was previously part of the actual screen delete pop over, if the `pageName` is null then I do:
```
if (pageName == null) {
      pageName = state.pages.main.name
    }
```

in the case of backend code for deleting a table we don't set the page name so it always uses main instead. For the screen popover it needs to be able to delete pages from the `unauthenticated` page so I had to have the page name in for that.

Reply:
No worries, if that's null when it's used then all good.